test/test_mypy.py

```diff
@@ -20,14 +20,40 @@
 from typing import TYPE_CHECKING, Any, Dict, Iterable, Iterator, List
 
 try:
-    from typing_extensions import TypedDict
+    if TYPE_CHECKING:
```

Comment:
Per our conversation yesterday, we should set all of these to `dict` when not type-checking to avoid the import error nesting.